### PR TITLE
Create tmp dir for go test

### PR DESCRIPTION
**Pull request by yutongz**
Didn't create tmp dir so that it failed to create test files

### CODE REVIEW

File: toolbox/pkg_check/main_test.go [hunk 1]
```diff
@@ -149,10 +149,21 @@ func TestFailedCheck(t *testing.T) {
 }
 
 func TestMain(m *testing.M) {
+	if _, err := os.Stat(tmpDir); os.IsNotExist(err) {
```
Comment (sebastienvas) [inline toolbox/pkg_check/main_test.go]: You might want to use https://golang.org/pkg/io/ioutil/#TempFile to make your test more predictable. As of now if I run tests concurrently they will fail. You want to make sure that each file name is unique.

**Review by sebastienvas — CHANGES_REQUESTED** [review]
PTAL

**Comment by yutongz** [comment]
PTAL

**Review by sebastienvas — APPROVED** [review]

File: toolbox/pkg_check/main_test.go [hunk 2]
```diff
@@ -149,10 +153,17 @@ func TestFailedCheck(t *testing.T) {
...
-	os.Exit(m.Run())
+	var err error
+	if tmpDir, err = ioutil.TempDir("", tmpDirPrefix); err != nil {
+		log.Printf("Failed to create tmp directory: %s, %s", tmpDir, err)
+		os.Exit(4)
+	}
+
+	exitCode := m.Run()
+
+	if err := os.RemoveAll(tmpDir); err != nil {
```
Comment (nclando) [inline toolbox/pkg_check/main_test.go:164]: why move this out of a defer?

Comment (yutongz) [reply toolbox/pkg_check/main_test.go:164]: cause os.Exit() will end it immediately without going through defer.

**Pull request merged**